functions/go/set-image/image_transformer_test.go

```diff
@@ -19,20 +19,31 @@ func itemsToYaml(items []*sdk.KubeObject) string {
 	return result
 }
 
-func runImageTransformerResults(input, config string) (*sdk.ResourceList, error) {
-	rl, err := sdk.ParseResourceList([]byte(input))
+func runImageTransformerResults(input, config string) (*fn.ResourceList, error) {
+	rl, err := fn.ParseResourceList([]byte(input))
 	if err != nil {
 		return nil, err
 	}
-	functionConfig, err := sdk.ParseKubeObject([]byte(config))
+	functionConfig, err := fn.ParseKubeObject([]byte(config))
 	if err != nil {
 		return nil, err
 	}
 	rl.FunctionConfig = functionConfig
-
-	if err = setImageTags(rl); err != nil {
+	in, _ := rl.ToYAML()
+	out, err := fn.Run(fn.ResourceListProcessorFunc(setImageTags), in)
```

Comment:
What is the motivation for changing the main entry point in this test scaffolding?

Reply:
Calling `SetImageTags` directly won't go through the SDK panic handling process. In fact, when using sdk methods, we expects users to either use `fn.AsMain` or `fn.Execute` which they will pass in the `fn.ResourceListProcessorFunc(setImageTags)` and not calling the method `setImageTags` directly